File: src/main/java/com/dogGetDrunk/meetjyou/config/CorsConfig.java
package com.dogGetDrunk.meetjyou.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.CorsConfigurationSource;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;

import java.util.List;

@Configuration
public class CorsConfig {

    @Value("${dns.url}")
    private String dnsUrl;

    @Bean
    public CorsConfigurationSource corsConfigurationSource() {
        CorsConfiguration config = new CorsConfiguration();

        config.setAllowedOrigins(List.of(dnsUrl));

        // 허용할 HTTP 메서드
        config.setAllowedMethods(List.of("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS"));

        // HTTP 요청 헤더 허용
        config.setAllowedHeaders(List.of("*"));

        // 요청이 Credentials (쿠키, Authorization 헤더 등)을 포함할 수 있도록 허용
        config.setAllowCredentials(true);

        // 모든 엔드포인트에 대해 CORS 설정 적용
        UrlBasedCorsConfigurationSource source = new UrlBasedCorsConfigurationSource();
        source.registerCorsConfiguration("/**", config);

        return source;
    }
}

File: src/main/java/com/dogGetDrunk/meetjyou/config/CorsConfig.kt
package com.dogGetDrunk.meetjyou.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.env.Environment
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.CorsConfigurationSource
import org.springframework.web.cors.UrlBasedCorsConfigurationSource

@Configuration
class CorsConfig(
    @Value("\${dns.url}")
    private val dnsUrl: String,
    private val environment: Environment,
) {
    @Bean
    fun corsConfigurationSource(): CorsConfigurationSource {
        val config = CorsConfiguration()

        // localhost origins are only needed for local development against a deployed backend;
        // allowing them in release would let any locally-run browser page make credentialed calls.
        config.allowedOriginPatterns = if (environment.activeProfiles.contains("dev")) {
            listOf(dnsUrl, "http://localhost:*", "http://127.0.0.1:*")
        } else {
            listOf(dnsUrl)
        }

        config.allowedMethods = listOf("GET", "POST", "PUT", "PATCH", "DELETE", "OPTIONS")
        config.allowedHeaders = listOf("*")
        config.allowCredentials = true

        val source = UrlBasedCorsConfigurationSource()
        source.registerCorsConfiguration("/**", config)

        return source
    }
}
